### PR TITLE
chore: Increase snowflake-connector-python dependency to 3.7

### PR DESCRIPTION
**Pull request by bushwhackr**
**What this PR does / why we need it**: change in snowflake signature `fetch_arrow_all` in snowflake-connector-python 3.7. 
This fixes linting issues and forces users to use the updated method.

Refer to #3952 and upstream fix: https://github.com/snowflakedb/snowflake-connector-python/issues/1800

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/infra/offline_stores/snowflake.py [hunk 1]
```diff
@@ -470,20 +470,9 @@ def _to_df_internal(self, timeout: Optional[int] = None) -> pd.DataFrame:
         return df
 
     def _to_arrow_internal(self, timeout: Optional[int] = None) -> pyarrow.Table:
-        pa_table = execute_snowflake_statement(
+        return execute_snowflake_statement(
             self.snowflake_conn, self.to_sql()
-        ).fetch_arrow_all(force_return_table=False)
```
Comment (tokoko) [inline sdk/python/feast/infra/offline_stores/snowflake.py:475]: This was a bug, right? Wouldn't this already raise an exception for <3.7 versions?

**Review by tokoko — APPROVED** [review]

**Comment by bushwhackr** [comment]
@shuchu @sudohainguyen if you could approve this fix

**Review by sudohainguyen — APPROVED** [review]
Lgtm

**Comment by bushwhackr** [comment]
@sudohainguyen Could you add the okay-to-test label, just to make sure that all the integration tests works.

**Pull request merged**